sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -477,11 +477,16 @@ abstract class GpuToTimestamp
     } else { // Timestamp or DateType
       lhs.getBase.asTimestampMicroseconds()
     }
-    withResource(tmp) { r =>
-      // The type we are returning is a long not an actual timestamp
-      withResource(Scalar.fromInt(downScaleFactor)) { downScaleFactor =>
-        withResource(tmp.asLongs()) { longMicroSecs =>
-          longMicroSecs.div(downScaleFactor)
+    // Return Timestamp value if dataType it is expecting is of TimestampType
+    if (dataType.equals(TimestampType)) {
+      tmp
+    } else {
+      withResource(tmp) { r =>
```

Comment:
nit: I don't see this `r` variable being used anywhere although it looks like `tmp` is referenced instead. This isn't incorrect but makes the code harder to read IMO. Perhaps just rename `r` to `tmp`?